codalab/worker/worker.py

```diff
@@ -168,6 +168,23 @@ def load_state(self):
                 resources=RunResources.from_dict(run_state.resources),
             )
 
+    def sync_state(self):
+        """
+        Sync worker run state by appending all additional new fields in the current RunState object
+        with the default value "None" to an older RunState object that is read from worker-state.json.
+        """
+        for uuid, run_state in self.runs.items():
+            values = [getattr(run_state, name) for name in run_state._fields]
+            old_len = len(run_state._fields)
+            new_len = len(RunState._fields)
+            # When there are additional new fields detected, recreate the run_state object
+            # to include those new fields with default "None" specified from RunState object
+            if old_len < new_len:
+                for i in range(old_len, new_len):
+                    values.append(None)
```

Comment:
Does this depend on the order of the field names of `RunState` defined in `worker_run_state.py`?

Reply:
Would be good to write this in a way that looks at the field names rather than assuming size and order are meaningful.

Reply:
Sure. I originally thought these existing fields are needed to keep the current worker functional, and appending new fields at the end seems to be reasonable, so I compare the length and keep the order of the existing run state object. But looking for field names is definitely more robust in case we want to change the list in the future.